input-explainer.md

```diff
@@ -176,16 +176,30 @@ Most applications will want to visually represent the input sources somehow. The
   * `'tracked-pointer'`: If the `gripTransform` is not `null` an application-appropriate controller model should be drawn using that transform. If appropriate for the experience, the a visualization of the target ray and a cursor as described in the `'gaze'` should also be drawn.
   * `'screen'`: In all cases the point of origin of the target ray is obvious and no visualization is needed.
 
+Determining what to render is important as well. In some cases it's appropriate for the application to render a contextually appropriate model (such as a racket in a tennis game), in which case the physical appearance of the input source doesn't matter much. Many other times, though, it's desirable for the application to display a device that matches what the user is holding, especially when relaying instructions about it's use. Finally there are times when it's best to not render anything at all, such as when the XR device uses a transparent display and the user can see their hands and/or any tracked devices without app assistance.
+
+ The `XRInputSource`'s `renderId` is used to determine what should be rendered in the cases when an application specific model isn't being used. 
+
+  - A `renderId` of `''` (empty string) indicates the input source should not be rendered. Cursors and pointing rays should still be rendered as described above if needed by the application. Input sources with a `targetRayMode` of `'gaze'` or `'screen'` MUST have a `renderId` of `''`. `'tracked-pointer'` input sources should typically only have a `renderId` of `''` when using displays that allow the user to see their environment and the physical input source.
+  - The `renderId` may be `'unknown'` if the type of input source cannot be reliably identified or the UA determines that the input source type must be masked for any reason.
+  - Inline sessions MUST only expose `renderId`s of `''` or `'unknown'`.
+  - Otherwise the `renderId` should be a lower-case string that describes the physical input source. It should not include and indication of the handedness of the input source, as that should be implied by the `handedness` attribute.
+    - If the input source is a tracked hand without any held controller, the `renderId` MUST be `hand`
```

Comment:
> it seems like the renderId should always have a valid id when possible because it will likely be what developers use to give instructions to users on how to use the experience

This is where `renderId` gets scary again - we're trying to contain it to the "what do I render?" question through its naming... but if it fills an unmet need in the spec (distinguishing hands from controllers), the attribute is likely to be abused elsewhere.

Another option would be to establish a separate `XRInputSourceType` enum and with explicit types like `"hand"` and `"controller"` to help reduce misuse of this attribute.

Reply:
A Leap Motion device attached to the front of a VR headset is an example of a case where I would expect that a `renderId` of `hand` would be appropriate. (And it would nicely pair with an `XRHand` interface as well.)